### PR TITLE
fix compiler invocations

### LINKED ISSUE #824: Clean up compiler invocations
This is just horrible:
`cd /home/zuntrax/git/openage/.bin/clang++-debug-Oauto-sanitize-none/openage/convert/interface && /usr/bin/clang++   -Dopenage_convert_interface_visgrep_EXPORTS -I/usr/include/python3.6m -I/usr/lib/python3.6/site-packages/numpy/core/include  -std=c++14 -g -O0 -fPIC    -DNDEBUG -fwrapv -O3  -Wno-#warnings  -std=c++14 -include "/home/zuntrax/git/openage/libopenage/pyinterface/hacks.h" -std=gnu++14 -o CMakeFiles/openage_convert_interface_visgrep.dir/visgrep.cpp.o -c /home/zuntrax/git/openage/openage/convert/interface/visgrep.cpp`

And this has problems, too:
`cd /home/zuntrax/git/openage/.bin/clang++-debug-Oauto-sanitize-none/libopenage && /usr/bin/clang++   -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_QML_LIB -DQT_QUICK_LIB -Dlibopenage_EXPORTS -I/usr/include/freetype2 -I/usr/include/opus -I/usr/include/SDL2 -isystem /usr/include/qt -isystem /usr/include/qt/QtCore -isystem /usr/lib/qt/mkspecs/linux-g++ -isystem /usr/include/qt/QtQuick -isystem /usr/include/qt/QtQml -isystem /usr/include/qt/QtNetwork -isystem /usr/include/qt/QtGui  -std=c++14 -g -O0 -fPIC    -Wall -Wextra -pedantic -Wno-gnu-statement-expression -fPIC -std=gnu++14 -o CMakeFiles/libopenage.dir/engine.cpp.o -c /home/zuntrax/git/openage/libopenage/engine.cpp`

Double `-std=c++14`, an additional `-std=gnu++14`, `-O0` and `-O3` in the same invocation, double `-fPIC`.
The build was configured as debug. Debug builds should default to `-Og` instead of `-O0`.

### PR DESCRIPTION
**Pull request by zuntrax**
fixes #824 
Maybe test with an ancient compiler needing `-std=gnu++1y`.
Debug and release compiler invocations for hand-written as well as cython generated files have been inspected. Most stuff is fixed. We still have double `-fPIC`, but that's fine.

### CODE REVIEW

File: buildsystem/HandlePythonOptions.cmake [hunk 1]
```diff
@@ -7,17 +7,10 @@ find_package(Python 3.4 REQUIRED)
 find_package(Cython 0.25 REQUIRED)
 find_package(Numpy REQUIRED)
 
-py_get_config_var(OPT PYEXT_CXXFLAGS)
```
Comment (TheJJ) [inline buildsystem/HandlePythonOptions.cmake]: depending on the platform, this would yield important flags that we otherwise could not anticipate. so i wouldn't remove it.

Comment (zuntrax) [inline buildsystem/HandlePythonOptions.cmake]: That's just the compiler flags, I don't think there will be a lot of fancy stuff on other platforms. The python include paths and all the other stuff we query the interpreter for is not affected by this change.

File: buildsystem/modules/FindPython.cmake [hunk 2]
```diff
@@ -182,7 +182,7 @@ foreach(PYTHON ${PYTHON_INTERPRETERS})
 		"${CMAKE_BINARY_DIR}"
 		SOURCES "${CMAKE_CURRENT_LIST_DIR}/FindPython_test.cpp"
 		LINK_LIBRARIES "${PYTHON_LIBRARY_NAME}"
-		CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${PYTHON_INCLUDE_DIR}" "-DLINK_DIRECTORIES=${PYTHON_LIBRARY_DIR}"
+		CMAKE_FLAGS "-DINCLUDE_DIRECTORIES=${PYTHON_INCLUDE_DIR}" "-DLINK_DIRECTORIES=${PYTHON_LIBRARY_DIR}" "-DCMAKE_CXX_STANDARD=14"
```
Comment (TheJJ) [inline buildsystem/modules/FindPython.cmake]: why is this necessary? aren't the cflags set already?

File: buildsystem/HandlePythonOptions.cmake [hunk 3]
```diff
@@ -29,9 +22,6 @@ if("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
 	set(PYEXT_CXXFLAGS "${PYEXT_CXXFLAGS} -Wno-#warnings")
 endif()
 
-# add our own regular C++ flags
-set(PYEXT_CXXFLAGS "${PYEXT_CXXFLAGS} ${CMAKE_CXX_FLAGS}")
```
Comment (TheJJ) [inline buildsystem/HandlePythonOptions.cmake]: where do the regular cxxflags get added then? or are they just used by default?

File: buildsystem/HandlePythonOptions.cmake (hunk 1 above)
Comment (TheJJ) [inline buildsystem/HandlePythonOptions.cmake]: alright, that's good.

File: buildsystem/HandlePythonOptions.cmake (hunk 3 above)
Comment (zuntrax) [inline buildsystem/HandlePythonOptions.cmake]: Jup, CMake adds them for every CXX target.

File: buildsystem/modules/FindPython.cmake (hunk 2 above)
Comment (zuntrax) [inline buildsystem/modules/FindPython.cmake]: Nope, that is happening in a dynamically generated CMake project. `CXX_STANDARD` and some other stuff do not get handed through. This is already fixed upstream, but those changes aren't on my system yet.
